theano/compile/builders.py

```diff
@@ -133,6 +135,59 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def connection_pattern(self, node):
```

Comment:
I am really not sure what you are referring to. The connection_pattern() method is not about computing shapes but rather knowing (mainly for gradient reasons) what inputs the value of each output depends on.

The connection pattern in ScanOp can be found here : https://github.com/Theano/Theano/blob/master/theano/scan_module/scan_op.py#L1539. The first thing it does is call inner_connection_pattern() which infers the connection pattern of the inner graph (exactly the same thing as you are doing).